spec/classes/rabbitmq_spec.rb

```diff
@@ -1788,12 +1788,87 @@
+      # other quorum_membership_reconciliation paramaters are set at all
+      # This ensures full backward compatibility with PRE RabbitMQ 3.13
+      describe 'rabbitmq-quorum_membership_reconciliation_enabled undef options' do
+        let(:params) { { quorum_membership_reconciliation_enabled: :undef } }
+
+        it 'sets quorum_membership_reconciliation_enabled parameter undef in config file' do
+          is_expected.not_to contain_file('rabbitmq.config'). \
+            with_content(%r{\{quorum_membership_reconciliation_enabled, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_auto_remove, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_interval, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_trigger_interval, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_target_group_size, .*\}})
+        end
+      end
+
+      # Ensure that whenever Param quorum_membership_reconciliation_enabled is false - none of the
+      # other quorum_membership_reconciliation paramaters are set at all
+      # This ensures full backward compatibility with PRE RabbitMQ 3.13
+      describe 'rabbitmq-quorum_membership_reconciliation_enabled false options' do
+        let(:params) { { quorum_membership_reconciliation_enabled: false } }
+
+        it 'sets quorum_membership_reconciliation_enabled parameter false in config file' do
+          is_expected.not_to contain_file('rabbitmq.config'). \
+            with_content(%r{\{quorum_membership_reconciliation_enabled, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_auto_remove, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_interval, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_trigger_interval, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_target_group_size, .*\}})
+        end
+      end
+
+      # Ensure that whenever Param quorum_membership_reconciliation_enabled is true - the defaults
+      # of all other quorum_membership_reconciliation parameters ensure they are UNSET until
+      # explicitly set.
+      describe 'rabbitmq-quorum_membership_reconciliation_enabled true options' do
+        let(:params) { { quorum_membership_reconciliation_enabled: true } }
+
+        it 'sets quorum_membership_reconciliation_enabled parameter true in config file' do
+          is_expected.to contain_file('rabbitmq.config'). \
+            with_content(%r{\{quorum_membership_reconciliation_enabled, true\}})
+        end
+
+        it 'sets quorum_membership_reconciliation_enabled parameter true in config file ONLY' do
+          is_expected.not_to contain_file('rabbitmq.config'). \
+            with_content(%r{\{quorum_membership_reconciliation_auto_remove, .*\}}). \
+            with_content(%r{\{quorum_membership_reconciliation_interval, .*\}}). \
```

Comment:
Done. I'm still learning "spec" tests overall (my only other experience is with inspec from chef a bit). Didn't know you could string them all in a single stanza like that - thank you!!

I removed the .*\} regex portion - but left the ", " portion in the matcher. My brain tells me this might filter out an improper match on an erroneous value or something at some point in the future by keeping the match more "specific"

Overall there's 3 test cases now comprising 3 sections in total, it definitely looks cleaner than it did, i think.